### PR TITLE
Add a targeted script to generate and copy a sqlmap command

### PR DESCRIPTION
**Pull request by juliosmelo**
It is a simple targeted command thats generate and copy a sqlmap command based on the request

### CODE REVIEW

**Comment by thc202** [comment]
- The changelog should be updated and the script name should be camel case: https://github.com/zaproxy/community-scripts/blob/main/CONTRIBUTING.md#contributing-to-community-scripts
- Commit needs to be signed off: https://github.com/zaproxy/community-scripts/pull/318/checks?check_run_id=12316105549

File: targeted/SQLMapCommandGenerator.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//author: @juliosmelo
+
+
+function invokeWith(msg) {
+	var string = "sqlmap --url '"+msg.getRequestHeader().getURI().toString()+"' \\\n";
+	var header = msg.getRequestHeader().getHeadersAsString();
+	header = header.split(msg.getRequestHeader().getLineDelimiter());
+
+	for(var i=0;i<header.length;i++){
+		//blacklisting Host (other blacklisting should also specify here
```
Comment (thc202) [inline targeted/SQLMapCommandGenerator.js:13]:
```suggestion
		// deny listing Host (other deny listing should also specify here)
```

File: targeted/SQLMapCommandGenerator.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		var keyval = header[i].split(":");
+		if(keyval[0].trim() != "Host"){
+			string += " -H '"+header[i].trim()+"' ";
+		}
+		if (keyval[0].trim() == "Host"){
```
Comment (thc202) [inline targeted/SQLMapCommandGenerator.js:18]:
```suggestion
		} else {
```

File: targeted/SQLMapCommandGenerator.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+function invokeWith(msg) {
+	var string = "sqlmap --url '"+msg.getRequestHeader().getURI().toString()+"' \\\n";
+	var header = msg.getRequestHeader().getHeadersAsString();
+	header = header.split(msg.getRequestHeader().getLineDelimiter());
+
+	for(var i=0;i<header.length;i++){
+		//blacklisting Host (other blacklisting should also specify here
+		var keyval = header[i].split(":");
+		if(keyval[0].trim() != "Host"){
+			string += " -H '"+header[i].trim()+"' ";
```
Comment (thc202) [inline targeted/SQLMapCommandGenerator.js:16]: ` '"+header[i].trim()+"' ";` could be done after the if/else, it's being done in both branches.

**Comment by thc202** [comment]
This needs to be rebased.

File: targeted/SQLMapCommandGenerator.js (hunk 3 above)
Comment (juliosmelo) [reply targeted/SQLMapCommandGenerator.js]: Hey, @thc202  thanks for the feedback. It's done

File: targeted/SQLMapCommandGenerator.js (hunk 1 above)
Comment (thc202) [reply targeted/SQLMapCommandGenerator.js]: This was done partially.

File: targeted/SQLMapCommandGenerator.js (hunk 3 above)
Comment (thc202) [reply targeted/SQLMapCommandGenerator.js]: The changes don't seem to be correct, as is now it's sending `Host: Host: localhost:9000`. The else should be added/kept to either add `--host` or `-H`, not both when the `Host` header. Or just remove the extra Host header handling, `sqlmap` send the host header also when specified with `-H`.

**Comment by thc202** [comment]
The previous comment still needs to be addressed (the changelog needs to be updated, i.e. move the new entry to the Unreleased section).

File: targeted/SQLMapCommandGenerator.js (hunk 3 above)
Comment (thc202) [reply targeted/SQLMapCommandGenerator.js]: Removed the host header handling.

**Comment by thc202** [comment]
Rebased to update the changelog.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Comment by juliosmelo** [comment]
Thank you !

**Pull request merged**